### PR TITLE
DOCSP-33546 DBTools 100.9.0 RN

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION
- Adds 100.9.0 changelog to release notes 

## STAGING
https://preview-mongodbajhuhmdb.gatsbyjs.io/database-tools/DOCSP-33546-100.9.0-RN/release-notes/database-tools-changelog/#100.9.0-changelog

## JIRA
https://jira.mongodb.org/browse/DOCSP-33546

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=654a59ad5b686d6bd9095ba1

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/changelogs/releases/100.9.0.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.. _100.9.0-changelog:
+
+100.9.0 Changelog
+-----------------
+
+*Released 2023-10-20*
+
+This version adds support for MongoDB Server 7.0 and updates the Go version 
+used to build the tools to Go 1.20.10.
```
Comment (ianf-mongodb) [inline source/includes/changelogs/releases/100.9.0.rst:9]: Suggestion, add link to the go minor release for additional context:
```
`Go 1.20.10<https://go.dev/doc/devel/release#go1.20.minor>`__
```

Comment (ianf-mongodb) [inline source/includes/changelogs/releases/100.9.0.rst:9]: Consider converting this to two separate bullets ([borrowed some verbiage from 100.7.5 for consistency](https://github.com/mongodb/docs-commandline-tools/blob/master/source/includes/changelogs/releases/100.7.5.rst)) :

```suggestion
- This version adds support for :manual:`MongoDB Server 7.0 </release-notes/7.0>`.
- This version updates the Go version used to build the tools `1.20.10<https://go.dev/doc/devel/release#go1.20.minor>`__ .
```

**Review by ianf-mongodb — APPROVED** [review]
Hi @ajhuh-mdb, thanks for picking this up, LGTM with optional changes.

**Review by tfogo — APPROVED** [review]
LGTM!

**Pull request merged**